packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts

```diff
@@ -99,46 +140,54 @@ export const getEventTypesFromGroup = async ({ ctx, input }: GetByViewerOptions)
     eventTypes.push(...teamEventTypes);
   }
 
-  let nextCursor: typeof cursor | undefined = undefined;
-  if (eventTypes && eventTypes.length > limit) {
+  let nextCursor: number | null | undefined = undefined;
+  if (eventTypes.length > limit) {
     const nextItem = eventTypes.pop();
     nextCursor = nextItem?.id;
   }
 
   const mappedEventTypes = await Promise.all(eventTypes.map(mapEventType));
 
-  log.info(
-    "mappedEventTypes before filtering",
+  log.debug(
+    "fetchEventTypesBatch",
     safeStringify({
-      input,
       mappedEventTypes,
     })
   );
 
-  const filteredEventTypes = mappedEventTypes.filter((eventType) => {
-    const isAChildEvent = eventType.parentId;
-    if (!isAChildEvent) {
+  return { eventTypes: mappedEventTypes, nextCursor: nextCursor ?? undefined };
+};
+
+const filterEventTypes = async (
+  eventTypes: MappedEventType[],
+  userId: number,
+  shouldListUserEvents: boolean | undefined,
+  teamId: number | null | undefined
+) => {
+  const filteredEventTypes = eventTypes.filter((eventType) => {
+    if (!eventType.parentId) {
       return true;
     }
     // A child event only has one user
     const childEventAssignee = eventType.users[0];
-    if (!childEventAssignee || childEventAssignee.id != ctx.user.id) {
+
+    if (!childEventAssignee || childEventAssignee.id !== userId) {
       return false;
     }
     return true;
   });
 
   log.info(
```

Comment:
@Udit-takkar we want this in .info or .debug like the rest?

Reply:
.info is fine